test/e2e/karmadactl_test.go

```diff
@@ -1373,6 +1376,196 @@ var _ = framework.SerialDescribe("Karmadactl taint testing", ginkgo.Labels{NeedC
 	})
 })
 
+var _ = ginkgo.Describe("Karmadactl apply testing", func() {
+	var (
+		member1Name        string
+		member1Client      kubernetes.Interface
+		deploymentManifest string
+		deployment         *appsv1.Deployment
+	)
+
+	ginkgo.BeforeEach(func() {
+		member1Name = framework.ClusterNames()[0]
+		member1Client = framework.GetClusterClient(member1Name)
+	})
+
+	ginkgo.BeforeEach(func() {
+		deployment = helper.NewDeployment(testNamespace, deploymentNamePrefix+rand.String(RandomStrLength))
+		deploymentManifest = fmt.Sprintf("/tmp/%s.yaml", deployment.Name)
+		err := WriteYamlToFile(deployment, deploymentManifest)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			os.Remove(deploymentManifest)
+		})
+	})
+
+	ginkgo.It("should apply configuration without propagating into member clusters", func() {
+		// Apply configuration without propagation to member clusters.
+		cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "apply", "-f", deploymentManifest)
+		output, err := cmd.ExecOrDie()
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		gomega.Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("deployment.apps/%s created", deployment.Name)))
+		defer framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
+
+		// Check that the deployment is created on karmada control plane.
+		framework.WaitDeploymentFitWith(kubeClient, deployment.Namespace, deployment.Name, func(_ *appsv1.Deployment) bool {
+			return true
+		})
+
+		// Check that no deployment is propagated to member1 cluster.
+		framework.WaitClusterFitWith(controlPlaneClient, member1Name, func(_ *clusterv1alpha1.Cluster) bool {
+			_, err = member1Client.AppsV1().Deployments(deployment.Namespace).Get(context.TODO(), deployment.Name, metav1.GetOptions{})
+			return apierrors.IsNotFound(err)
+		})
```

Comment:
Is this check not needed, because we have no intention of distributing resources?

Reply:
Great point! After reviewing the `karmadactl apply --help` command, I discovered that `karmadactl apply` is inherited from `kubectl apply`, which is not the intended behavior for these test cases.

```bash
$ karmadactl apply --help
...
# Apply the configuration without propagation into member clusters. It acts as 'kubectl apply'.
kubectl karmada apply -f manifest.yaml
...
```